src/components/table/template.test.js

```diff
@@ -0,0 +1,374 @@
+/* globals describe, it, expect */
+
+const { render, getExamples } = require('../../../lib/jest-helpers')
+
+const examples = getExamples('table')
+describe('Table', () => {
+
+  it('renders with classes', () => {
+    const $ = render('table', {
+      'classes': 'custom-class-goes-here',
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+
+    expect($component.hasClass('custom-class-goes-here')).toBeTruthy()
+  })
+
+  it('renders with caption text', () => {
+    const $ = render('table', examples['table-with-caption-and-head'])
+    const $caption = $('.govuk-c-table__caption')
+
+    expect($caption.text()).toBe('Caption 1 : Months and rates')
+  })
+
+  it('renders the caption size class', () => {
+    const $ = render('table', examples['table-with-caption-and-head'])
+    const $caption = $('.govuk-c-table__caption')
+
+    expect($caption.hasClass('small')).toBeTruthy()
+  })
+
+  it('renders first cell in every row as a <th> element with correct `govuk-c-table__header` class and `scope=row` attribute', () => {
+    const $ = render('table', {
+      'firstCellIsHeader': true,
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          },
+          {
+            'text': '£95',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£75',
+            'format': 'numeric'
+          },
+          {
+            'text': '£55',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    })
+    const $component = $('.govuk-c-table')
+    const $tableBody = $component.find('.govuk-c-table .govuk-c-table__body')
+    const $firstTableRow = $tableBody.find('.govuk-c-table__row:first-child')
+    const $firstTableHeader = $firstTableRow.find('.govuk-c-table__header:first-child')
+
+    expect($firstTableHeader.get(0).tagName).toEqual('th')
+    expect($firstTableHeader.attr('scope')).toEqual('row')
+    expect($firstTableHeader.hasClass('govuk-c-table__header')).toBeTruthy()
+
+
+    const $lastTableRow = $tableBody.find('.govuk-c-table__row:last-child')
+    const $lastTableHeader = $lastTableRow.find('.govuk-c-table__header:first-child')
+
+    expect($lastTableHeader.get(0).tagName).toEqual('th')
+    expect($lastTableHeader.attr('scope')).toEqual('row')
+    expect($lastTableHeader.hasClass('govuk-c-table__header')).toBeTruthy()
+
```

Comment:
_Block must not be padded by blank lines_

Reply:
fixed